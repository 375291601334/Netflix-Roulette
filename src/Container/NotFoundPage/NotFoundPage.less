@import '../../Styles/globalStyles.less';
@import '../../Styles/variables.less';

.topLogo {
  position: absolute;
  top: 20px;
  left: 100px;

  .tablet({
    left: 50px;
  });

  .mobile({
    left: 30px;
  });
}

.container {
  flex: auto;
  padding: 20px 100px 40px;
  display: flex;
  flex-direction: column;
  background: @darkGrey;
  color: @white;

  .tablet({
    padding: 20px 50px 30px;
  });

  .mobile({
    padding: 20px 30px 30px;
  });
}

.content {
  margin: auto;
}

.title {
  font-size: 50px;
  text-align: center;
}

.icon {
  display: block;
  margin: -40px auto;
  mask-image: url(../../../assets/icons/404.svg);
  -webkit-mask-image: url(../../../assets/icons/404.svg);
  width: 350px;
  height: 350px;
  background: linear-gradient(@black, @red, @white);

  .mobile({
    width: 100%;
  });
}

a {
  text-decoration: none;
}

.homeButton {
  display: block;
  box-sizing: border-box;
  padding: 20px;
  margin: 0 auto;
  font-size: 20px;
  font-weight: bold;
  border: 3px solid @grey;
  border-radius: 5px;
  background: none;
  color: @red;
  cursor: pointer;

  &:focus,
  &:hover {
    background: @grey;
    filter: contrast(190%);
  }
}
