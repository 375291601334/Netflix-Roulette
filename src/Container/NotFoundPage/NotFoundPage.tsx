import React from 'react';
import Link from 'next/link';
import { Logo, Footer } from '../../Components';
import css from './NotFoundPage.less';

export function NotFoundPage() {
  return (
    <>
      <Logo className={css.topLogo} />
      <div className={css.container}>
        <div className={css.content}>
          <h2 className={css.title}>Page Not Found</h2>
          <span className={css.icon} />
          <Link href="/films" >
            <button className={css.homeButton}>GO BACK TO HOME</button>
          </Link>
        </div>
      </div>
      <Footer>
        <Logo />
      </Footer>
    </>
  );
}
