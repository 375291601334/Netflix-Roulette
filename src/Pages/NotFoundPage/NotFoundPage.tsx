import React from 'react';
import { Link } from 'react-router-dom';
import { Footer, Logo } from '../../Components';
import css from './NotFoundPage.less';

export function NotFoundPage() {
  return (
    <div className={css.container}>
      <div className={css.content}>
        <h2 className={css.title}>Page Not Found</h2>
        <span className={css.icon} />
        <Link to={'/films'} >
          <button className={css.homeButton}>GO BACK TO HOME</button>
        </Link>
      </div>
    </div>
  );
}
