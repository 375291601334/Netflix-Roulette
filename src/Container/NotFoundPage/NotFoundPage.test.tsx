import React from 'react';
import { render } from '@testing-library/react';
import { NotFoundPage } from './NotFoundPage';

describe('NotFoundPage', () => {
  test('render NotFoundPage', () => {
    const { baseElement } = render(<NotFoundPage />);
    expect(baseElement).toMatchSnapshot();
  });
});
