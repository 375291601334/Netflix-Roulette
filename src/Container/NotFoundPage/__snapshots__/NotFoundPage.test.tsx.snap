// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage render NotFoundPage 1`] = `
<body>
  <div>
    <div
      class="logo topLogo"
    >
      <span>
        netflix
      </span>
      roulette
    </div>
    <div
      class="container"
    >
      <div
        class="content"
      >
        <h2
          class="title"
        >
          Page Not Found
        </h2>
        <span
          class="icon"
        />
        <button
          class="homeButton"
        >
          GO BACK TO HOME
        </button>
      </div>
    </div>
    <footer>
      <div
        class="logo undefined"
      >
        <span>
          netflix
        </span>
        roulette
      </div>
    </footer>
  </div>
</body>
`;
